oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/ResourceUtils.java

```diff
@@ -58,12 +60,39 @@ public static InputStream readToStream(String fileName) throws FileNotFoundExcep
             throw new FileNotFoundException("path not found: " + parentPath);
         }
         final Set<String> nameSet = new HashSet<>(Arrays.asList(fileNames));
-        final File[] listFiles = Objects.requireNonNull(new File(url.getPath())
-            .listFiles((dir, name) -> nameSet.contains(name)), "No files in " + parentPath);
+        final File[] listFiles = Objects.requireNonNull(
+            new File(url.getPath())
+                .listFiles((dir, name) -> nameSet.contains(name)), "No files in " + parentPath);
 
         if (listFiles.length == 0) {
             throw new FileNotFoundException("files not found:" + nameSet);
         }
         return listFiles;
     }
+
+    public static List<File> getPathFilesRecursive(String path) throws FileNotFoundException {
```

Comment:
Notice, this kind of `Recursive` could easily cause OOM. Like I commented https://github.com/apache/skywalking/pull/7130#discussion_r654422789, please make the structure solid.

Reply:
We should have depth as a parameter, as in this OpenAPI case, we only need recursive depth as `1`, which means current folder and one more(service name folder).

Reply:
done